org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/desktop/TabbedDesktop.java

```diff
@@ -76,6 +78,8 @@ public ProcessDialog openProcessDialog(int processId, boolean soTrx) {
 			pd.setTitle(null);
 			preOpenNewTab();
 			windowContainer.addWindow(tabPanel, title, true, null);
+			ProcessDialogOnCloseHandler handler = new ProcessDialogOnCloseHandler();
```

Comment:
this can be safer and simplify by passing m_windowno from pd to handler, i.e new ProcessDialogOnCloseHandler(pd.getWindowNow());

Reply:
hmm .. actually, ProcessDialog already override the onPageDetached method, this changes can probably be reduce to just a one liner addition to the ProcessDialog.onPageDetached method - SessionManager.getAppDesktop().unregisterWindow(m_m_WindowNo)

Reply:
Thanks @hengsin - tested your suggestion and it works fine - is committed now